hydrostats/ens_metrics.py

```diff
@@ -409,6 +409,17 @@ def ens_crps(obs, fcst_ens, adj=np.nan, remove_neg=False, remove_zero=False, llv
     return output
 
 
+def ens_crps_decomposition(obs, fcst_ens, adj=np.nan, remove_neg=False, remove_zero=False):
+
+    # Calculate the Psam quantity
+    Psam = fcst_ens - obs               # Difference the bins
+    Psam[Psam < 0] = 0                  # Apply the heavyside function
+    Psam = np.sum(Psam, axis=0)         # Aggregate across the ensembles
+
+    # Calculate the Pcli quantity
+
+
+
 @jit(nopython=True, parallel=True)
 def numba_crps(ens, obs, rows, cols, col_len_array, sad_ens_half, sad_obs, crps, adj):
```

Comment:
The rest of the changes in this function look ok but i'm not an expert in crps. @waderoberts123 could you check this part?